minimal-plugin/src/main/java/io/micronaut/gradle/MicronautMinimalApplicationPlugin.java

```diff
@@ -102,14 +103,17 @@ private void configureJavaExecTasks(Project project, Configuration developmentOn
                     Map<String, Object> sysProps = new LinkedHashMap<>();
                     sysProps.put("micronaut.io.watch.restart", true);
                     sysProps.put("micronaut.io.watch.enabled", true);
-                    javaExec.doFirst(workaroundEagerSystemProps -> {
-                        String watchPaths = sourceSet
-                                .getAllSource()
-                                .getSrcDirs()
-                                .stream()
-                                .map(File::getPath)
-                                .collect(Collectors.joining(","));
-                        javaExec.systemProperty("micronaut.io.watch.paths", watchPaths);
+                    javaExec.doFirst(new Action<Task>() {
```

Comment:
Add a comment above this code to prevent this code being refactored back into a lambda